### ISSUE TITLE
Memory usage rapidly climbs when compiled with tempo tracker.

### ISSUE THREAD

**Issue by jgshaban** [BarkyTheDog/catclock#4]
Hi there!

I found your repo a few days ago. This little applet is really fun to have on the desktop! I keep grinning like a dope when I sneak a glance at it while I'm working.

As much fun as it is to keep running and ticking to the beat in the corner of my workspace, I noticed a pretty big memory usage problem after letting it run in the background for a few days.

Big as in ~47GB big. I only noticed when a machine with 64GB started swapping.

I noted down as much as I can at the moment.

- Leaking only occurs when the tempo tracker is enabled at compile time.
- Initial execution with no options takes around 30MB when compiled with tempo tracker.
- Without tempo tracker, memory usage is ~8MB, and stays there.
- Tested with both Ubuntu and Arch systems with the same symptoms.
- Ram usage climbs by ~1MB/6-10s

If you need more information, please let me know.

Thanks a bunch!

EDIT: Corrected leak rate from 5-7s to 6-10s

**Comment by sashakoshka** [comment]
This seems to originate in TempoTrackerThread, where an infinite loop reads incoming audio data. At the start of the loop, it puts a buffer of size BUFSIZE onto the stack, and then allocates an fvec_t of size BUFSIZE, and then replaces the pointer to its data with the buffer on the stack. The allocated buffer in the fvec_t is never freed.

**Comment by jgshaban** [comment]
I'm sorry I took so long to give any feedback. Life things happened, and I left this account alone for a very long time.

I grabbed another copy of the source, and compiled it with the tempo tracker enabled. Afterwards, I left it running for a few days in the background during regular operations.

I can confirm that as of right now, it's sitting at a nice, tidy, 34MB. It hasn't climbed at all in the past two days

Looks like the commit from @sashakoshka did the trick!

I think it's safe to say that this can be closed.
